File: src/problems/leetcode-cn/3634. 使数组平衡的最少移除数目/solution/python/solution.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        length = len(nums)
        if length < 2:
            return 0

        nums.sort()
        l, r, inc = 0, 1, 0

        while r < length:
            if nums[r] <= nums[l]*k:
                inc = max(inc, r-l+1)
                r += 1
            else:
                l += 1
        
        return length-inc

File: src/problems/leetcode-cn/3634. 使数组平衡的最少移除数目/solution/go/solution.go
func minRemoval(nums []int, k int) int {
    length := len(nums)
    if length < 2 {
        return 0
    }

    sort.Ints(nums)

    // 滑动窗口
    l, r, inc := 0, 1, 0
    for r < length {
        if nums[r] <= nums[l]*k {
            inc = max(inc, r-l+1)
            r++
        } else {
            l++
        }
    }

    return length - inc
}